pygeo/parameterization/DVGeoCST.py

```diff
@@ -0,0 +1,1093 @@
+
+        # Check that the leading edge is at y = 0
+        idxLE = np.argmin(self.foilCoords[:, self.xIdx])
+        yLE = self.foilCoords[idxLE, self.yIdx]
+        if abs(yLE) > 1e-2:
+            raise ValueError(f"Leading edge y (or idxVertical) value must equal zero, not {yLE}")
+
+        # Determine the trailing edge thickness
+        idxTE = np.where(self.foilCoords[:, self.xIdx] == self.xMax)[0]
+        self.yUpperTE = np.max(self.foilCoords[idxTE, self.yIdx])
+        self.yLowerTE = np.min(self.foilCoords[idxTE, self.yIdx])
+        self.thicknessTE = self.yUpperTE - self.yLowerTE
+
+        # Compute splines for the upper and lower surfaces (used to split the foil in addPointSet)
+        self.foil = Airfoil(coords)
+        self.upperSpline, self.lowerSpline = self.foil.splitAirfoil()
+
+        # Fit CST parameters to the airfoil's upper and lower surface
+        self.idxFoil = {}
+        self.idxFoil["upper"], self.idxFoil["lower"] = self._splitUpperLower(self.foilCoords)
+        self.defaultDV["chord"][0] = self.xMax - self.xMin
+        for dvType in ["upper", "lower"]:
+            self.defaultDV[dvType] = self.computeCSTfromCoords(
+                self.foilCoords[self.idxFoil[dvType], self.xIdx],
+                self.foilCoords[self.idxFoil[dvType], self.yIdx],
+                self.defaultDV[dvType].size,
+                N1=self.defaultDV[f"n1_{dvType}"],
+                N2=self.defaultDV[f"n2_{dvType}"],
+                dtype=self.dtype,
+            )
+
+    def addPointSet(self, points, ptName, **kwargs):
+        """
+        Add a set of coordinates to DVGeometry
+        The is the main way that geometry in the form of a coordinate list is given to DVGeometry
+        to be manipulated.
+        This assumes...
+            - Trailing edge is vertical or sharp and at maximum x values
+            - The geometry is exclusively an extruded shape (no spanwise changes allowed)
+            - The airfoil's leading edge is on the left (min x) and trailing edge is
+              on the right (max x)
+            - The airfoil's leading edge is at y (or idxVertical) equals zero (within :math:`10^{-2}`)
+
+        Parameters
+        ----------
+        points : array, size (N,3)
+            The coordinates to embed. These coordinates *should* all project into the interior of the geometry.
+        ptName : str
+            A user supplied name to associate with the set of coordinates.
+            This name will need to be provided when updating the coordinates or when
+            getting the derivatives of the coordinates.
+        kwargs
+            Any other parameters are ignored.
+        """
+        # Convert points to the type specified at initialization (with isComplex) and store the points
+        points = points.astype(self.dtype)
+
+        # Check that all points are within the airfoil x bounds
+        if np.any(points[:, self.xIdx] < self.xMin) or np.any(points[:, self.xIdx] > self.xMax):
```

Comment:
Shouldn't we allow small deviations? This can be a kwarg, and has to do with the accuracy of your mesh and how well it fits the airfoil coordinates.